lib/bigshift/cli.rb

```diff
@@ -25,12 +25,22 @@ def initialize(argv, options={})
     end
 
     def run
-      setup
-      unload
-      transfer
-      load
-      cleanup
-      nil
+      begin
+        setup
+        unload
+        transfer
+        load
+        cleanup
+        nil
+      rescue NoMethodError => e
+        if ['access_key_id', 'match'].any? { |key| e.message.include? key }
+          raise CliError.new('AWS configuration missing or malformed: ' + e.message, e.backtrace, @usage)
+        else
+          raise CliError.new(e.message, e.backtrace, @usage)
```

Comment:
I mentioned in another comment (it was a thread from the last review, it doesn't show in the diff any longer, unfortunately) that I would like this to not catch `NoMethodError` – but if it turns out that it will have to do that for some reason, this branch should not translate all other `NoMethodError`s to `CliError`. This should just `raise`. The other branch catches an expected error and translates it into a more semantic error, but this branch is for the unexpected errors and they should not be masked as something which they aren't.